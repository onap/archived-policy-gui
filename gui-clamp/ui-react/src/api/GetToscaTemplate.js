/*-
 * ============LICENSE_START=======================================================
 *  Copyright (C) 2021 Nordix Foundation.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 * ============LICENSE_END=========================================================
 */
import React, { useState } from "react";
import Button from "react-bootstrap/Button";

const GetToscaTemplate = (props) => {

  const [windowLocationPathName, setWindowLocationPathname] = useState('');

  const getTemplateHandler = async () => {
    console.log('getTemplateHandler called')
    setWindowLocationPathname(window.location.pathname);

    const params = {
      name: props.templateName,
      version: props.templateVersion
    }

    const response = await fetch(windowLocationPathName +
      '/restservices/clds/v2/toscaControlLoop/getToscaTemplate' + '?' + (new URLSearchParams(params)));

    const data = await response.json();

    props.onGetToscaServiceTemplate(data);

  }

  return (
    <React.Fragment>
      <Button variant="primary"
              type="submit"
              onClick={ getTemplateHandler }>Get Tosca Service Template</Button>
    </React.Fragment>
  );


}

export default GetToscaTemplate;
