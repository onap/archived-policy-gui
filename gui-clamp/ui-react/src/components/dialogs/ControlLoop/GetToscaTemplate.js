/*
 * ============LICENSE_START=======================================================
 *  Copyright (C) 2021 Nordix Foundation.
 *  ================================================================================
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 *
 *  SPDX-License-Identifier: Apache-2.0
 *  ============LICENSE_END=========================================================
 */

import React, { useState } from "react";
import Button from "react-bootstrap/Button";
import ControlLoopService from "../../../api/ControlLoopService";

const GetToscaTemplate = (props) => {

  const getTemplateHandler = async () => {

    const response = await ControlLoopService.getToscaTemplate(props.templateName, props.templateVersion)
      .catch(error => error.message);

    if(!response.ok) {
      console.log('getToscaServiceTemplateHandler called with error')

    } else
    {
      console.log('getToscaServiceTemplateHandler called')
    }

    props.onGetToscaServiceTemplate(response);

  }

  return (
    <React.Fragment>
      <Button variant="primary"
              type="submit"
              onClick={ getTemplateHandler }>Pull Tosca Service Template</Button>
    </React.Fragment>
  );


}

export default GetToscaTemplate;
