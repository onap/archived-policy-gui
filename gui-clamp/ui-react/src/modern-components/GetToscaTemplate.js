import React from "react";
import Button from "react-bootstrap/Button";

const GetToscaTemplate = (props) => {

  const getTemplateHandler = async () => {

    const params = {
      name: props.templateName,
      version: props.templateVersion
    }

    const response = await fetch('https://localhost:3000' +
      '/onap/controlloop/v2/commission/' + 'toscaservicetemplate'
      + '?' + (new URLSearchParams(params)), {
      headers: {
        'Authorization': 'Basic ' + btoa('healthcheck' + ':' + 'zb!XztG34')
      }
    });

    const data = await response.json();

    props.onGetToscaServiceTemplate(data);

  }

  return (
    <React.Fragment>
      <Button variant="primary"
              type="submit"
              onClick={ getTemplateHandler }>Get Tosca Service Template</Button>
    </React.Fragment>
  );


}

export default GetToscaTemplate;