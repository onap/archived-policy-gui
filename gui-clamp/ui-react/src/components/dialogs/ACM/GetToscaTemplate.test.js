/*
 * ============LICENSE_START=======================================================
 *  Copyright (C) 2021 Nordix Foundation.
 *  ================================================================================
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 *
 *  SPDX-License-Identifier: Apache-2.0
 *  ============LICENSE_END=========================================================
 */

import React from 'react';
import { mount, shallow } from 'enzyme';
import GetToscaTemplate from './GetToscaTemplate';
import toJson from "enzyme-to-json";
import { act } from "react-dom/test-utils";

describe('Verify GetToscaTemplate', () => {

  const flushPromises = () => new Promise(setImmediate);

  it("renders without crashing", () => {
    shallow(<GetToscaTemplate/>);
  });

  it("renders correctly", () => {
    const tree = shallow(<GetToscaTemplate/>);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('should have a Button element', () => {
    const container = shallow(<GetToscaTemplate/>)
    expect(container.find('Button').length).toEqual(1);
  });

  it('button should call getTemplateHandler when clicked when response is error', async () => {

    const onGetToscaServiceTemplate = jest.fn()
    jest
      .spyOn(global, 'fetch')
      .mockImplementationOnce(async () =>
        Promise.resolve({
            ok: false,
            status: 200,
            json: () => {
              return Promise.resolve({
                "tosca_definitions_version": "tosca_simple_yaml_1_1_0",
                "data_types": {},
                "policy_types": {},
                "topology_template": {},
                "name": "ToscaServiceTemplateSimple",
                "version": "1.0.0",
                "metadata": {},
                "id": "0.19518677404255147"
              })
            }
          }
        )
      )

    const component = mount(<GetToscaTemplate onGetToscaServiceTemplate={onGetToscaServiceTemplate}/>)
    const logSpy = jest.spyOn(console, 'log');

    await act(async () => {
      component.find('[variant="primary"]').simulate('click');
      await flushPromises()
      component.update()
      expect(logSpy).toHaveBeenCalledWith('getToscaServiceTemplateHandler called with error');
    });
    component.unmount();
  });

  it('should have a Button element with specified text', () => {
    const container = shallow(<GetToscaTemplate/>)
    expect(container.find('Button').text()).toBe('Pull Tosca Service Template');
  });
});
