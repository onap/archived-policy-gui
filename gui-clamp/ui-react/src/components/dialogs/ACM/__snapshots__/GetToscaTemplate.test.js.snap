// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify GetToscaTemplate renders correctly 1`] = `
<Fragment>
  <Button
    active={false}
    disabled={false}
    onClick={[Function]}
    type="submit"
    variant="primary"
  >
    Pull Tosca Service Template
  </Button>
</Fragment>
`;
